# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/webscheduling/WebTaskSchedulerMainThread.h
# Commit: 4ccb6970887e
# Full Hash: 4ccb6970887ece212a1176681a567480ea586dfc
# Author: Sean Feng <sefeng@mozilla.com>
# Date: 2025-02-25 09:30:56
# Regressor Bug: 1920115
# File Overlap Count: 2
# Description:
#   Bug 1920115 - Allow high priority tasks to run before timers r=smaug
#   
#   In Gecko, unlike other runnables where the TaskController
#   picks one each at a time, the TimeoutManager allows multiple
#   timers to run sequentially without allowing other runnables
# ==============================================================================

diff -r 7522970c5a1d -r 4ccb6970887e dom/webscheduling/WebTaskSchedulerMainThread.h
--- a/dom/webscheduling/WebTaskSchedulerMainThread.h	Mon Feb 24 19:31:28 2025 +0000
+++ b/dom/webscheduling/WebTaskSchedulerMainThread.h	Mon Feb 24 19:31:28 2025 +0000
@@ -32,6 +32,9 @@
   explicit WebTaskSchedulerMainThread(nsIGlobalObject* aParent)
       : WebTaskScheduler(aParent) {}
 
+  void IncreaseNumNormalOrHighPriorityQueuesHaveTaskScheduled() override;
+  void DecreaseNumNormalOrHighPriorityQueuesHaveTaskScheduled() override;
+
  private:
   nsresult SetTimeoutForDelayedTask(WebTask* aTask, uint64_t aDelay,
                                     EventQueuePriority aPriority) override;